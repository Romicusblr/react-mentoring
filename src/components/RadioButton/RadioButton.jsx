/* eslint-disable jsx-a11y/label-has-for */
import React from 'react';
import PropTypes from 'prop-types';
import style from './RadioButton.module.css';

function RadioButton({ value, name, defaultChecked }) {
  const radioId = `${name}_${value}`;
  return (
    <p className={style.radioButton}>
      <input type="radio" name={name} id={radioId} defaultChecked={defaultChecked} />
      <label htmlFor={radioId}>{value}</label>
    </p>
  );
}

RadioButton.propTypes = {
  value: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  defaultChecked: PropTypes.bool,
};

RadioButton.defaultProps = {
  defaultChecked: false,
};

export default RadioButton;
