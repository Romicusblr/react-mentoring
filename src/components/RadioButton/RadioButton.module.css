.radioButton {
  &>input {
    appearance: none;
    &:checked + label {
      background-color: var(--color-raised);
    }
  }

  &>label {
    padding: .5rem 1.5rem;
    cursor: pointer;
    background-color: var(--color-secondary);
  }

  &:first-of-type>label {
    border-radius: .25rem 0 0 .25rem;
  }

  &:last-of-type>label {
    border-radius: 0 .25rem .25rem 0 ;
  }
}
