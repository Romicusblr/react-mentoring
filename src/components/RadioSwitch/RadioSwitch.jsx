import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import style from './RadioSwitch.module.css';

function RadioSwitch({
  title, children, className, switchValue, ...other
}) {
  return (
    <div
      className={classNames(style.radioSwitch, className)}
      {...other}
    >
      <span>{title}</span>
      {React.Children.map(children, (child) => {
        const defaultChecked = switchValue === child.props.value;
        return React.cloneElement(child, { defaultChecked, ...other });
      })}
    </div>
  );
}

RadioSwitch.propTypes = {
  children: PropTypes.node.isRequired,
  title: PropTypes.string.isRequired,
  className: PropTypes.string,
  switchValue: PropTypes.string.isRequired,
};

RadioSwitch.defaultProps = {
  className: '',
};

export default RadioSwitch;
