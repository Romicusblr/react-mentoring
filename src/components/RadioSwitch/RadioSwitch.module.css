.radio {
  text-transform: uppercase;
  display: flex;
  flex-direction: row;
  align-items: center;
  border: none;
  padding: 0;
  margin: 0;

  &>span {
    margin-right: 1rem;
  }
  &>p {
    &>input {
      appearance: none;
      &:checked + label {
        background-color: var(--color-raised);
      }
    }

    &>label {
      padding: .5rem 1.5rem;
      cursor: pointer;
    }

    &:first-of-type>label {
      border-radius: .25rem 0 0 .25rem;
    }

    &:last-of-type>label {
      border-radius: 0 .25rem .25rem 0 ;
    }
  }

}
