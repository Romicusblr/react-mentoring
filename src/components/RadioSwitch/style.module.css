.radio {
  text-transform: uppercase;
  display: flex;
  align-items: center;

  &>span {
    margin-right: 1rem;
  }
  &>p {
    display: inline-block;
    &>input {
      appearance: none;
      &:checked + label {
        background-color: var(--color-pink);
      }
    }
    &>label {
      background-color: var(--color-gray);
      padding: .5rem 1.5rem;
      cursor: pointer;
    }

    &:first-of-type>label {
      border-radius: .25rem 0 0 .25rem;
    }

    &:last-of-type>label {
      border-radius: 0 .25rem .25rem 0 ;
    }
  }
}
