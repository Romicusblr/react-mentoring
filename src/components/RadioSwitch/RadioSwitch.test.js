import RadioSwitch from '.';
import RadioButton from '../RadioButton';

describe('RadioSwitch', () => {
  it('should match to snapshot', () => {
    const snap = shallow(
      <RadioSwitch title="radio">
        <RadioButton value="value" />
      </RadioSwitch>,
    );
    expect(snap).toMatchSnapshot();
  });
});
