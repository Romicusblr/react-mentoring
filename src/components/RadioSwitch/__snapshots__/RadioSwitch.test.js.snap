// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioSwitch should match to snapshot 1`] = `
<div
  className="radioSwitch"
>
  <span>
    radio
  </span>
  <RadioButton
    className=""
    defaultChecked={false}
    key=".0"
    name="radio"
    value="value"
  />
</div>
`;
